spock-specs/src/test/groovy/org/spockframework/smoke/extension/TempDirExtensionSpec.groovy

```diff
@@ -0,0 +1,124 @@
+package org.spockframework.smoke.extension
+
+import org.spockframework.EmbeddedSpecification
+import spock.lang.Shared
+import spock.lang.Stepwise
+import spock.lang.TempDir
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+/**
+ * @author dqyuan
+ */
+@Stepwise
+class TempDirExtensionSpec extends EmbeddedSpecification {
+
+  @Shared
+  @TempDir
+  Path sharedDir
+  @Shared
+  Path previousShared
+
+  @TempDir
+  File iterationDir
+  @Shared
+  File previousIteration
+
+  @TempDir
+  def untypedPath
+
+  def "temp dir exist"() {
+    previousShared = sharedDir
+    previousIteration = iterationDir
+
+    expect:
+    Files.exists(sharedDir)
+    iterationDir.exists()
+    iterationDir.directory
+    untypedPath instanceof Path
+  }
+
+  def "@TempDir creates only one dir for one spec, if the annotated field is 'shared'"() {
+    expect:
+    sharedDir == previousShared
+  }
+
+  def "get a different path for non-shared field, and previous was deleted"() {
+    expect:
+    iterationDir != previousIteration
+    !previousIteration.exists()
+  }
+
+  def "@TempDir creates one temp directory per iteration for normal field"() {
+    expect:
+    iterationDir != previousIteration
+
+    cleanup:
+    previousIteration = iterationDir
+
+    where:
+    i << (1..3)
+  }
+
+  def "have unwritable directory in temp directory"(int i) {
+    expect:
+    if (i == 0) {
+      def dir = Paths.get(iterationDir.toURI())
+      def aaa = dir.resolve("aaa")
+      def aaabbb = aaa.resolve("bbb")
+      def tempFile = aaabbb.resolve("tmp.txt")
+      Files.createDirectories(aaabbb)
+      Files.write(tempFile, "ewfwf".getBytes())
+      aaabbb.toFile().setWritable(false)
+      aaa.toFile().setWritable(false)
+      tempFile.toFile().setWritable(false)
+      previousIteration = iterationDir
+    } else if (i == 1) {
+      assert !previousIteration.exists()
+    }
+
+    where:
+    i << [0, 1]
+  }
+
+  static Path pathFromEmbedded
+
+  def "define temp directory location and keep temp directory using using configuration script"() {
+    given:
+    def userDefinedBase = Paths.get("build")
+    runner.configurationScript {
+      tempdir {
+        baseDir userDefinedBase
+        keep true
+      }
+    }
+    runner.addClassImport(Path)
+    runner.addClassImport(TempDir)
```

Comment:
This should be unnecessary, everything in `spock.lang` is imported automatically in the embedded spec runner

Reply:
```
startup failed:
script15974219807271941296144.groovy: 1: unable to resolve class Path 
 @ line 1, column 76.
   ASpec extends Specification { @TempDir
```

When I delete this line, this test will report above error.

Reply:
No, it will report so if you remove the line with `Path` and that is what it complains about with "unable to resolve class Path".
`TempDir` is automatically imported, as it is in the `spock.lang` package.
(I also tested it before I wrote that comment as I was only 98% sure. ;-) )

Reply:
Got it